app/config.go

```diff
@@ -411,12 +411,10 @@ func (s *Server) SaveConfig(newCfg *model.Config, sendConfigChangeClusterMessage
 		return model.NewAppError("saveConfig", "app.save_config.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
 
-	if s.Metrics != nil {
-		if *s.Config().MetricsSettings.Enable {
-			s.Metrics.StartServer()
-		} else {
-			s.Metrics.StopServer()
-		}
+	if *s.Config().MetricsSettings.Enable {
```

Comment:
should we check the `s.startMetrics` attribute here too?

Reply:
Should we call the `s.Metrics.Register` function too? what happens if I have disabled the metrics at start time but I enable it later?

Reply:
Same thoughts.

Reply:
Yes we should use `s.startMetrics` and also +1 to registering handlers here.